pkg/clusterresource/builder.go

```diff
@@ -158,9 +155,7 @@ func (o *Builder) Build() ([]runtime.Object, error) {
 
 	var allObjects []runtime.Object
 	allObjects = append(allObjects, o.generateClusterDeployment())
-	if !o.SkipMachinePoolGeneration {
-		allObjects = append(allObjects, o.generateMachinePool())
-	}
+	allObjects = append(allObjects, o.generateMachinePool())
```

Comment:
Let's put a nil check on the generated machine pool so that we don't have to add back the `SkipMachinePoolGeneration` when another platform comes along for which we don't yet have machine pool support.

Reply:
I'm not sure what you man here. `o.generatedMachinePool` doesn't seem like it will ever return a nil.

Can you explain your thoughts further here please?

Reply:
For the platforms that we have now, `generateMachinePool` does not return a nil. If we add support for a new platform for which Hive does not support MachinePools yet, then we need a way to skip machine pool generation. This should not come from a flag on the options: It should come from the builder implementation being used.

I am fine with leaving it as is for this PR and dealing with it when we do add support for the next platform.

Reply:
Ok, sounds good. We'll look at this in a future PR.